backend/test/v2/integration/upgrade_test.go

```diff
@@ -0,0 +1,601 @@
+	_, err := s.runClient.Create(createRunRequest)
+	require.Nil(t, err)
+}
+
+func (s *UpgradeTests) VerifyRuns() {
+	t := s.T()
+
+	/* ---------- List the runs, sorted by creation time ---------- */
+	runs, _, _, err := test.ListRuns(
+		s.runClient,
+		&runParams.ListRunsParams{SortBy: util.StringPointer("created_at")},
+		s.resourceNamespace)
+	require.Nil(t, err)
+	assert.True(t, len(runs) >= 1)
+	assert.Equal(t, "hello world", runs[0].DisplayName)
+
+	/* ---------- Get hello world run ---------- */
+	helloWorldRunDetail, err := s.runClient.Get(&runParams.GetRunParams{RunID: runs[0].RunID})
+	require.Nil(t, err)
+	assert.Equal(t, "hello world", helloWorldRunDetail.DisplayName)
+	assert.Equal(t, "this is hello world", helloWorldRunDetail.Description)
+}
+
+func (s *UpgradeTests) PrepareRecurringRuns() {
+	t := s.T()
+
+	pipeline := s.getHelloWorldPipeline(true)
+	experiment := s.getHelloWorldExperiment(true)
+
+	/* ---------- Create a new hello world job by specifying pipeline ID ---------- */
+	createRecurringRunRequest := &recurring_run_params.CreateRecurringRunParams{Body: &recurring_run_model.V2beta1RecurringRun{
+		DisplayName: "hello world",
+		Description: "this is hello world",
+		PipelineVersionReference: &recurring_run_model.V2beta1PipelineVersionReference{
+			PipelineID:        pipeline.PipelineID,
+			PipelineVersionID: pipeline.PipelineVersionID,
+		},
+		ExperimentID:   experiment.ExperimentID,
+		MaxConcurrency: 10,
+		Mode:           recurring_run_model.RecurringRunModeENABLE,
+		NoCatchup:      true,
+	}}
+	_, err := s.recurringRunClient.Create(createRecurringRunRequest)
+	require.Nil(t, err)
+}
+
+func (s *UpgradeTests) VerifyRecurringRuns() {
+	t := s.T()
+
+	pipeline := s.getHelloWorldPipeline(false)
+	experiment := s.getHelloWorldExperiment(false)
+
+	/* ---------- Get hello world recurring run ---------- */
+	recurringRuns, _, _, err := test.ListAllRecurringRuns(s.recurringRunClient, s.resourceNamespace)
+	require.Nil(t, err)
+	require.Len(t, recurringRuns, 1)
+	recurringRun := recurringRuns[0]
+
+	expectedRecurringRun := &recurring_run_model.V2beta1RecurringRun{
+		RecurringRunID: recurringRun.RecurringRunID,
+		DisplayName:    "hello world",
+		Description:    "this is hello world",
+		ExperimentID:   experiment.ExperimentID,
+		PipelineVersionReference: &recurring_run_model.V2beta1PipelineVersionReference{
+			PipelineID:        pipeline.PipelineID,
+			PipelineVersionID: pipeline.PipelineVersionID,
+		},
+		ServiceAccount: test.GetDefaultPipelineRunnerServiceAccount(*isKubeflowMode),
+		MaxConcurrency: 10,
+		NoCatchup:      true,
+		Mode:           recurring_run_model.RecurringRunModeENABLE,
+		Namespace:      recurringRun.Namespace,
+		CreatedAt:      recurringRun.CreatedAt,
+		UpdatedAt:      recurringRun.UpdatedAt,
+		Trigger:        recurringRun.Trigger,
+		Status:         recurringRun.Status,
+	}
+
+	assert.Equal(t, expectedRecurringRun, recurringRun)
+}
+
+func (s *UpgradeTests) VerifyCreatingRunsAndJobs() {
```

Comment:
nit: function name should use recurring run

Reply:
Good catch. Fixed.